### ISSUE TITLE
Your http://magnum-project.net/ is down

### ISSUE THREAD

**Issue by waynebrantley** [phatboyg/Magnum#24]

**Comment by phatboyg** [comment]
Yeah, I didn't renew the domain name, the site will be rehosted on GitHub at some point.

**Comment by togakangaroo** [comment]
any news on this? I'm trying to trace through Mass Transit examples and that leads me to this project.

**Comment by redwards510** [comment]
I was hoping for some kind of description of what this project does.

**Comment by phatboyg** [comment]
Well, the readme on the GitHub site has a brief summary. It's a kitchen sink.

https://github.com/phatboyg/Magnum

However, Magnum has been deprecated from MassTransit v3 forward, and the Magnum State Machine is deprecated in favor of Automatonymous.

**Comment by snekbaev** [comment]
The latest MassTransit manual suggests to use ```CombGuid.Generate()``` to create clustered-index friendly Guids, however your last comment mentions that MT3 doesn't use Magnum anymore. I know it uses another library NewId for guids. Should I add Magnum reference or just use NewId's ```NewId.NextGuid()``` for CorrelationId?
```
Event( () => ItemAdded, x => x.CorrelateBy( cart => cart.UserName, context => context.Message.UserName ).SelectId( context => CombGuid.Generate()) );
```
or
```
Event( () => ItemAdded, x => x.CorrelateBy( cart => cart.UserName, context => context.Message.UserName ).SelectId( context => NewId.NextGuid()) );
```
Persisting sagas using EF. 

I can see the generated guids differences, though can't for sure say which one is better for SQL. One thing that I did is timed both:
```
void Main()
{
	var capLimit = 1000000;
	var d = new Dictionary<Guid, int>(capLimit);
	
	var sw = new Stopwatch();
	sw.Start();
	
	for (int i = 0; i < capLimit; i++)
	{
		d.Add(CombGuid.Generate(), 0);
	}

	sw.Stop();
	sw.Elapsed.Dump("First done in:");
	
	d.Clear();
	
	sw.Restart();

	for (int i = 0; i < capLimit; i++)
	{
		d.Add(NewId.NextGuid(), 0);
	}

	sw.Stop();
	sw.Elapsed.Dump("Second done in:");
}
```

Results:

```
First done in:
00:00:01.1889361 

Second done in:
00:00:15.3374765 
```
Apparently the generation speed has some significant difference on large volumes. When I set the cap limit to 200000, difference is quite small:
```
First done in:
00:00:00.2339395 

Second done in:
00:00:01.0658077 
```
And yes, I understand the test case vs (majority of) real life cases difference :) Just wanted to share some findings :)

**Comment by phatboyg** [comment]
Yes, you should use NewId. It has replaced CombGuid and is able to be used across multiple machines without collisions.

The ordering is also SQL friendly for clustered indices.

There shouldn't be a performance difference that matters, in fact, take out the dictionary and I think you'll find it is faster. The equality comparison of NewId might be causing the inserts to be slower.

**Comment by snekbaev** [comment]
I used dictionary to check whether it will crash or not, just wanted to make sure I get unique every time :) ATM I using SQL's newid() for PK + identity field for clustered index. Now I'm thinking to rework data access layer to pregenerate guids on insert using your NewId and get rid of identity column. Another reason why I doubted: I'm very-very new to MT and going through its examples. ShoppingCart demo has correlation id setup as:
```
Event( () => ItemAdded, x => x.CorrelateBy( cart => cart.UserName, context => context.Message.UserName ).SelectId( context => Guid.NewGuid() ) );
```
which is not SQL clustered-index friendly. Was wondering why, however now it's clear, I will use NewId library. Thank you!
